protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java

```diff
@@ -100,6 +100,7 @@ protected void doAuth(AuthenticationAttempt authenticationAttempt, ImapSession s
                     session.stopDetectingCommandInjection();
                 } catch (BadCredentialsException e) {
                     authFailure = true;
+                    LOGGER.info("User {} failed authentication", authenticationAttempt.getAuthenticationId().asString());
```

Comment:
Now can we base our selves on the audit trail to not add another log?

Reply:
Why is remoteIp empty?

Reply:
Because it's unavailable now. We need to change code to get remoteIP

Reply:
@thanhbv200585 

I suggest to expose remoteIP via `ImapSession` interface:
```java
    /**
     * Return the {@link InetSocketAddress} of the remote peer
     */
    InetSocketAddress getRemoteAddress();
```

Implement it in `NettyImapSession`:
```java
    @Override
    public InetSocketAddress getRemoteAddress() {
        return (InetSocketAddress) channel.remoteAddress();
    }
```

Then add remoteIP in `AbstractAuthProcessor`' AuditTrail (s).

Reply:
Please take also care of Proxy protocol!

Reply:
Not solved IMO and this log should not be merged, as it brings nothing than just making work some poc/demo... 

I thought we were supposed to try to use audit trail logs (already in place) for this?

Reply:
I think it makes sense, of course we need to add more some info such as remoteIP